strawberry/channels/testing.py

```diff
@@ -0,0 +1,102 @@
+import dataclasses
+import uuid
+from typing import AsyncIterator, Dict, List, Optional, Tuple, Union
+
+from graphql import GraphQLError
+
+from channels.testing.websocket import WebsocketCommunicator
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL
+from strawberry.subscriptions.protocols.graphql_transport_ws.types import (
+    ConnectionAckMessage,
+    ConnectionInitMessage,
+    SubscribeMessage,
+    SubscribeMessagePayload,
+)
+from strawberry.types import ExecutionResult
+
+
+class GraphQLWebsocketCommunicator(WebsocketCommunicator):
+    """
+    Usage:
+    ```python
+    import pytest
+    from strawberry.channels.testing import GqlWsCommunicator
+    from myapp.asgi import application
+
+
+    @pytest.fixture
+    async def gql_communicator():
+        async with GqlWsCommunicator(application, path="/graphql") as client:
+            yield client
+
+
+    async def test_subscribe_echo(gql_communicator):
+        async for res in gql_communicator.subscribe(
+            query='subscription { echo(message: "Hi") }'
+        ):
+            assert res.data == {"echo": "Hi"}
+    ```
+    """
+
+    def __init__(
+        self,
+        application,
```

Comment:
Could we add typings to `application`? I think it should be `ASGIApplication` from [asgiref](https://github.com/django/asgiref/blob/730d865506adf565d53e5a67cceb9f796dc472ba/asgiref/typing.py#L256)

Reply:
AFAIK `ASGIApplication` is a concept, and can't be typed. it can be a function that returns a function, it can be a class, etc...